### PR TITLE
Add more tests for mplot3d

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary`

Added tests for a bunch of mplot3d code.

~Removed `get_frame_on` and `set_frame_on` as they can be inherited (only difference is the . As I did not really see the effect of them, I also removed the use in the `draw` method. I tried to check back, but couldn't really figure out why it was forced to another value there. Seems a bit doubtful.~

~In the `margins` method, the logic for raising was changed. Currently, passing e.g. `margins(1, x=2)` was allowed, setting all the margins to 1 without any type of information. Now it raises instead. Not sure if that is the idea, but it seems to make more sense considering the error message and expected usage. Actually, the same thing happens in the standard `margins` method as well...~

https://github.com/matplotlib/matplotlib/blob/51ce6776b1a9089295652fe91567e0c814f3f95f/lib/matplotlib/axes/_base.py#L2721-L2723

## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [N/A] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by WeatherGod** [comment]
Is there a test that verifies turning the panes on and off works?

**Comment by oscargus** [comment]
> Is there a test that verifies turning the panes on and off works?

No, there isn't as far as I can tell. And it doesn't really seem like it does work (as far as I can tell). I will revert those changes and just keep the coverage increase. Will see if I can get that working (or at least test it) in another PR.

File: lib/mpl_toolkits/tests/test_mplot3d.py [hunk 1]
```diff
@@ -1643,6 +1659,83 @@ def test_computed_zorder():
...
+    assert ax.format_coord(0, 0) == 'x=1.8066, y=1.0367, z=−0.0553'
+
+
+def test_get_axis_position():
+    fig = plt.figure()
+    ax = fig.add_subplot(projection='3d')
+    x = np.arange(10)
+    ax.plot(x, np.sin(x))
+    fig.canvas.draw()
+    assert ax.get_axis_position() == (False, True, False)
```
Comment (QuLogic) [inline lib/mpl_toolkits/tests/test_mplot3d.py:1685]: If you understand what this does, it could do with a docstring. It appears unused though.

**Review by QuLogic — APPROVED** [review]

File: lib/mpl_toolkits/tests/test_mplot3d.py (hunk 1 above)
Comment (oscargus) [reply lib/mpl_toolkits/tests/test_mplot3d.py:1685]: I cannot really say that I do, no...

Comment (oscargus) [reply lib/mpl_toolkits/tests/test_mplot3d.py:1685]: Maybe it should be deprecated? This was deprecated as a renderer property in 3.4 and removed now: https://github.com/matplotlib/matplotlib/blob/main/doc/api/next_api_changes/removals/00001-DS.rst

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
The docs failure is the sphinx-gallery issue with resetting the units registry, merging over the failure.

**Pull request merged**